components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -0,0 +1,327 @@
+                                                   encoded_value.end());
+    result.insert(result.end(), encoded_value_bytes.begin(),
+                  encoded_value_bytes.end());
+  } else if (type == "bytes") {
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> bytes;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &bytes));
+    const std::vector<uint8_t> encoded_value = KeccakHash(bytes);
+    result.insert(result.end(), encoded_value.begin(), encoded_value.end());
+  } else if (type == "bool") {
+    absl::optional<bool> value_bool = value.GetIfBool();
+    if (!value_bool)
+      return absl::nullopt;
+    uint256_t encoded_value = (uint256_t)*value_bool;
+    for (int i = 256 - 8; i >= 0; i -= 8) {
+      result.push_back((encoded_value >> i) & 0xFF);
+    }
+  } else if (type == "address") {
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> address;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &address));
+    DCHECK_EQ(address.size(), 20u);
+    for (size_t i = 0; i < 256 - 160; i += 8)
+      result.push_back(0);
+    result.insert(result.end(), address.begin(), address.end());
+
+  } else if (base::StartsWith(type, "bytes", base::CompareCase::SENSITIVE)) {
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 5, &type_check) || type_check > 32)
+      return absl::nullopt;
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> bytes;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &bytes));
+    if (bytes.size() > 32)
+      return absl::nullopt;
+    result.insert(result.end(), bytes.begin(), bytes.end());
+    for (size_t i = 0; i < 32u - bytes.size(); ++i) {
+      result.push_back(0);
+    }
+  } else if (base::StartsWith(type, "uint", base::CompareCase::SENSITIVE)) {
+    // uint8 to uint256 in steps of 8
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 4, &type_check) || type_check % 8 ||
+        type_check > 256)
+      return absl::nullopt;
+
+    absl::optional<int> value_int = value.GetIfInt();
+    if (!value_int)
+      return absl::nullopt;
+    uint256_t encoded_value = *value_int;
+    // check if value excceeds type bound
+    switch (type_check) {
+      case 8:
+        if (encoded_value > std::numeric_limits<uint8_t>::max())
+          return absl::nullopt;
+        break;
+      case 16:
+        if (encoded_value > std::numeric_limits<uint16_t>::max())
+          return absl::nullopt;
+        break;
+      case 32:
+        if (encoded_value > std::numeric_limits<uint32_t>::max())
+          return absl::nullopt;
+        break;
+      default:
+        // TODO(darkdh): base::Value needs to support uint256 type
```

Comment:
Could we do something with Blob Values here like in RLP encoding we use `RLPUint256ToBlobValue`?

Reply:
Per Slack, the base::Value is read in by the JSON parser so we can't set it to a blob Value.

Reply:
nit: pls make comment say uint64_t uint128_t and uint256_t

Reply:
fixed in 6bbb1c4516cb09859c699807dc35b5fd17d06b54